### PR TITLE
 Website Interface Update - Round 4

### LINKED ISSUE #273: Consistently format internal links
Currently internal links are a mix between text and button links, these should be consistent.

A unit plan page could render all links within tables, for example. See #313.

### LINKED ISSUE #275: Add custom box link CSS
The CSS for box links (see homepage and topic pages) should be written from scratch as the existing solution doesn't size or wrap correctly.

- [ ] Link blocks should not have underline (hover or anytime).
- [ ] Link blocks should have animation for background change, and change to darker colour (rather than lighter). Same as panels.

### LINKED ISSUE #313: Use table to display list of unit plans, lessons, and programming exercises
Similar to the design of the table for curriculum integrations.

### LINKED ISSUE #319: Consistently name template files
Currently template files are a mix between dashes and underscores for separators, these should be consistent.

### LINKED ISSUE #329: Update curriculum areas and colours

### LINKED ISSUE #330: Improve design of resource pages

### PR DESCRIPTION
**Pull request by JackMorganNZ**
This pull request does the following:
- Consistently name template files (fixes #319)
- Improves design of resource home page (fixes #330)
- Updates curriculum areas and colours (fixes #329)
- Use tables to display list of unit plans, lessons, and programming exercises (fixes #313)
- Consistently formats internal links (fixes #273)
- Adds custom `box-link` CSS (fixes #275)

### CODE REVIEW

File: csunplugged/utils/group_lessons_by_age.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+"""Return ordered groups of lessons."""
+
+from collections import OrderedDict
+
+
+def group_lessons_by_age(lessons):
+    """Return ordered groups of lessons.
+
+    Lessons are grouped by the lesson minimum age and maximum ages,
+    and then order by number.
```
Comment (hayleyavw) [inline csunplugged/utils/group_lessons_by_age.py]: args?

File: csu [hunk 2]
```diff
@@ -176,7 +176,7 @@ function static_scratch {
 # Collecting static files
 function collect_static {
   printf "\nCollecting static files..."
-  docker-compose exec django /docker_venv/bin/python3 ./manage.py collectstatic --no-input
+  docker-compose exec django /docker_venv/bin/python3 ./manage.py collectstatic --no-input --clear
```
Comment (hayleyavw) [inline csu]: what is this wizardry? ``--clear``?

Comment (JackMorganNZ) [inline csu]: See: https://docs.djangoproject.com/en/1.11/ref/contrib/staticfiles/#cmdoption-collectstatic-clear

We wipe the `build` folder on each build of the static files, we probably should be wiping `staticfiles` too!

**Comment by JackMorganNZ** [comment]
@hayleyavw This is ready for checking again.

**Pull request merged**